extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,319 @@
+  }
+
+  @JsonProperty
+  public int getNominalEntries()
+  {
+    return nominalEntries;
+  }
+
+  @JsonProperty
+  public List<String> getMetricColumns()
+  {
+    return metricColumns;
+  }
+
+  @JsonProperty
+  public int getNumberOfValues()
+  {
+    return numberOfValues;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new CacheKeyBuilder(AggregatorUtil.ARRAY_OF_DOUBLES_SKETCH_CACHE_TYPE_ID).appendString(name)
+        .appendString(fieldName).appendInt(nominalEntries).appendStrings(metricColumns).appendInt(numberOfValues)
+        .build();
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return ArrayOfDoublesUnion.getMaxBytes(nominalEntries, numberOfValues);
+  }
+
+  public static final Comparator<ArrayOfDoublesSketch> COMPARATOR = new Comparator<ArrayOfDoublesSketch>()
+  {
+    @Override
+    public int compare(final ArrayOfDoublesSketch a, final ArrayOfDoublesSketch b)
+    {
+      return Doubles.compare(a.getEstimate(), b.getEstimate());
+    }
+  };
+
+  private static void updateUnion(final ArrayOfDoublesUnion union, final Object obj)
+  {
+    if (obj == null) {
+      return;
+    } else if (obj instanceof ArrayOfDoublesSketch) {
+      union.update((ArrayOfDoublesSketch) obj);
+    }
+    throw new IAE("Object of type [%s] can not be unioned", obj.getClass().getName());
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.<AggregatorFactory> singletonList(
+      new ArrayOfDoublesSketchAggregatorFactory(
+        fieldName,
+        fieldName,
+        nominalEntries,
+        metricColumns,
+        numberOfValues
+      )
+    );
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new ArrayOfDoublesSketchAggregatorFactory(name, name, nominalEntries, null, numberOfValues);
+  }
+
+  @Override
+  public Object finalizeComputation(final Object object)
+  {
+    return ((ArrayOfDoublesSketch) object).getEstimate();
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    if (metricColumns == null) {
+      return ArrayOfDoublesSketchModule.ARRAY_OF_DOUBLES_SKETCH_MERGE_AGG;
+    }
+    return ArrayOfDoublesSketchModule.ARRAY_OF_DOUBLES_SKETCH_BUILD_AGG;
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass().equals(o.getClass())) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+
+    final ArrayOfDoublesSketchAggregatorFactory that = (ArrayOfDoublesSketchAggregatorFactory) o;
+
+    if (!name.equals(that.name)) {
+      return false;
+    }
+    if (!fieldName.equals(that.fieldName)) {
+      return false;
+    }
+    if (nominalEntries != that.nominalEntries) {
+      return false;
+    }
+    if (!metricColumns.equals(that.metricColumns)) {
```

Comment:
Null-aware Objects.equals()

Reply:
Could you clarify please what do you think needs to be changed?

Reply:
Oh, I see, metricColumns can be null